server/routes/getOffences.ts

```diff
@@ -0,0 +1,32 @@
+import { Request, Response, NextFunction } from 'express';
+import { factory, formatFilename } from '../../shared/logging';
+
+const log = factory.getLogger(formatFilename(__filename));
+
+const getOffences = async (models, req: Request, res: Response, next: NextFunction) => {
+  const { chain, startDate, endDate } = req.query;
+
+  if (!chain) return next(new Error('Cannot find chain id'));
+
+  const where: any = {
+    '$ChainEventType.chain$': chain,
+    '$ChainEventType.event_name$': 'offence'
+  };
+  if (startDate && endDate) {
+    where.created_at = {
+      $between: [startDate, endDate]
+    };
+  }
+  const offences = await models.ChainEvent.findAll({
+    where,
+    include: [
+      { model: models.ChainEventType }
+    ]
+  });
+
+  if (!offences) return res.json({ status: 'Failure' }); // No Invites
```

Comment:
I would follow other patterns that are used in the routes when we throw errors. Namely, using `next(new Error(Errors.$ERROR_MESSAGE))`